File: LABS/Aquecimento.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool isMaiuscula(char c){
    return(c >= 'A' && c <= 'Z');
}

int count_M(char* palavra){
    int resp = 0;
    int tam = strlen(palavra);
    for(int i = 0; i < tam; i++ ){
        if(isMaiuscula(palavra[i])){
            resp++;
        }
    }

    return resp;
}


int main(){
    char entrada[100];
    
    while(true){
        scanf(" %[^\n]", entrada);
        if(strcmp(entrada, "FIM") == 0){
            break;
        }

        int resp = count_M(entrada);
        printf("%d\n", &resp);


    }

}

File: Aquecimento.java
import java.util.*;

public class Aquecimento {

    /*
     * Metodo para detectar letras Maisculas
     */
    public static boolean isMaiuscula(char c){
        return(c >= 'A' && c <= 'Z');
    }

    /*
     * Metodo para contar letras Maisculas recursivamente
     */

    static int count_M(String palavra){
        return count_M(palavra, 0, 0);
    }
    static int count_M(String palavra, int i, int count){
        if(i >= palavra.length()){
            return count;
        }
        if(isMaiuscula(palavra.charAt(i))){
            count++;

        }
       
        return count_M(palavra, i+1, count) ;
    }


   
    public static void main(String[] args){
       Scanner sc = new Scanner(System.in);

       String entrada;

       while(true){
        entrada = sc.nextLine();
        
        if(entrada.equals("FIM")){
            break;
        }

        System.out.println(count_M(entrada));

       }

       sc.close();


    }
    


}